### PR TITLE
Add markdownlint with GitHub actions

### PR DESCRIPTION
**Pull request by Youssef1313**
As the samples repository will start to have markdown files (for the samples browser) (and even it had md files before the samples browser), I think it's good to have a markdown linter as with dotnet/docs.

### CODE REVIEW

**Comment by NextTurn** [comment]
Why don't you submit all the changes in a single commit?

**Comment by Youssef1313** [comment]
> 
> 
> Why don't you submit all the changes in a single commit?

Working with GitHub interface. 😄
If this is a problem, I can switch to do that locally.

**Comment by NextTurn** [comment]
No problem. But Outlook won't group these mails into the same conversion for me.

image

**Comment by Youssef1313** [comment]
> 
> 
> No problem. But Outlook won't group these mails into the same conversion for me.
> 
> image

Hmm, I'm wondering why these emails were delivered to you at the first place.
Anyway, I'll switch to work locally and push one commit after I finish this.

File: windowsforms/datagridview/CSWinFormDataGridView/JustInTimeDataLoading/README.md [hunk 1]
```diff
@@ -1,39 +1,32 @@
...
 
 ## Code Logic
 
-1.  Enable VirtualMode on the DataGridView control by setting the VirtualMode
-    property to true:
+1. Enable VirtualMode on the DataGridView control by setting the VirtualMode property to true:
 
     ```CSharp
-    this.dataGridView1.VirtualMode = true
+    this.dataGridView1.VirtualMode = true;
```
Comment (Youssef1313) [inline windowsforms/datagridview/CSWinFormDataGridView/JustInTimeDataLoading/README.md]: nit: added a missing simi colon

**Comment by Youssef1313** [comment]
@NextTurn, I had to make some `git push`es while working locally when enabling markdown rules, to let the check tell me where are the violations. If you got these emails, hope they didn't bother you.

**Pull request merged**